### DISCUSSION TITLE
What forum should we prioritize for Q&A

### DISCUSSION THREAD

**Discussion by cmungall** [linkml/linkml#1828] [Q&A] [answered]
Right now we have different ways for the community to ask questions (as opposed to request concrete features or to report bugs, where the mechanism is always github issues)

- linkml-community on slack
- [here](https://github.com/orgs/linkml/discussions)
- [StackOverflow](https://stackoverflow.com/) using [this tag](https://stackoverflow.com/questions/tagged/linkml)
- our [Linkedin Group](https://www.linkedin.com/groups/14303246/)
- [@linkml_data on X](https://twitter.com/linkml_data)
- [@linkml on mastodon](https://qoto.org/@linkml) TODO: move from qoto
- email

We do not at this time have discord, zulip, discourse, ...

Currently we have a lot of great engagement on slack, but this is not ideal as the sole mechanism, we don't pay for it, so we got locked out of our own history. And the great Q&A there is not broadly visible (some people reportedly like this).

On the one hand we want to meet people where they are on the medium they are comfy with. On the other we want to ensure that things don't get dropped and that discussions are not spread or duplicated. And we want to preserve at least a subset of discussions where accepted answers are recorded for questions.

One possibility is that we keep with the status quo, but make more of an effort to preserve valuable Q&A pairs, either on the FAQ, or we make duplicate posts to SO, or on these discussion forums. Ideally with the SO approach the original questioner would repeat their question there and the original answerer would repeat their accepted answer.

This discussion is intended to be open ended about people's preferences and how we should manage this.

**Comment by rly** [comment]
As a developer, I prefer using Slack for more quick, informal, pseudo-private Q&A, and either GitHub discussions/issues or StackOverflow otherwise. My main entry point for finding information is googling the issue, and both GitHub discussions and StackOverflow are indexed by Google, so I go wherever I see a relevant Q&A. If I want to post a Q, I would probably pick GitHub discussions since I know I will reach the developers more easily.

I think the social media platforms are not great for Q&A - threading can be poor and IMO there is just not a lot of engagement there. For this kind of project, I think those are primarily useful for outreach/announcements.

**Comment by dalito** [comment]
GitHub for Q&A, issues, discussions. Slack (or discord) for quick questions, announcements or chats. The larger community (beyond just linkml) is nice on your Slack.

**Comment by WolfgangFahl** [comment]
IMHO GitHub and Stackoverflow should be used a lot more to gain public interest. Slack is somewhat a closed community. Also LLMs will not pick up the knowledge so e.g. ChatGPT4 hallucinates heavily on how LinkML might be used.

**Comment by cmungall** [comment, marked as answer]
Summary:

- Slack as the primary mechanism quick, informal, pseudo-private Q&A
    - `OBO#linkml-dev` for developers
    - `OBO#linkml-community` for users and general discussion
    - We may need to move off the OBO slack to our own workspace in the future, but it's a good home for now with many aligned people
- This discussion forum for indexed questions and discussion
- We should try and use [this tag](https://stackoverflow.com/questions/tagged/linkml) on SO more, both for Q and A
- social media for more general outreach and announcements (discussions welcome but may not be as well monitored)
    - [LinkML on Linkedin](https://www.linkedin.com/groups/14303246/)
    - [@linkml on mastodon](https://fosstodon.org/@linkml) (no longer on qoto - thanks @sneakers-the-rat and @cthoyt!)
    - [@linkml_data on X](https://twitter.com/linkml_data) de-prioritized
